src/test/java/org/opensearch/sdk/TestExtensionInterfaces.java

```diff
@@ -1,92 +0,0 @@
-/*
```

Comment:
This file should not have been deleted!

Reply:
May I just create a new pull request? I didn't know how to remove those files :(

Reply:
You can, but let's try something else first:
1. use `git reset --hard HEAD~2` to back up your repository to where it was before those last two deletions
2. change to the main branch, pull from upstream
3. go back to your develop branch and type `git rebase main`
4. force-push the results

Reply:
@dbwiddis, I've tried to do what you said. Does that work?

Reply:
Well those deleted files are back, which is good.   They are part of a merged PR #527 so I'm not sure the rebase worked as you wanted.  It's fine for now, it's just noise on the PR.  But a merge or rebase to main should fix it.

I'm not clear about your question on the XContentParser.  Is the code above taken from the test for the XContent parsing and you're trying to make the equivalent for the named writeables?

Reply:
@dbwiddis, Yes, that's what I am trying to do.

Reply:
The thing is that ToXContentObject, XContentParser, JsonXContent, XContentBuilder, etc. are taken from the library. Should I create equivalent for all of these name writeables(Example: WritableParser)? If yes, where should I put it? What folder?

Reply:
Not completely sure as I haven't dug into this but some quick observations/advice:
 - none of the XContent things are really relevant here.  writeable is a totally different implementation
 - there is a `NamedWriteableRegistryTests` class on OpenSearch that is probably useful here, but it really just shows a regisration.  It may be much simpler in your case as well.
 - bottom line: if you have a test that invokes all the new methods you are adding and produces expected results you should be good
 - you can run `gradle diffCoverage` to make sure you've covered the new code